components/worktemplates/components/customize.tsx

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import styled from 'styled-components';
+import {useIntl} from 'react-intl';
+
+import PublicPrivateSelector from 'components/widgets/public-private-selector/public-private-selector';
+import {Visibility} from '@mattermost/types/worktemplates';
+import Input from 'components/widgets/inputs/input/input';
+
+export interface CustomizeProps {
+    className?: string;
+    name: string;
+    visibility: Visibility;
+
+    onNameChanged: (name: string) => void;
+    onVisibilityChanged: (visibility: Visibility) => void;
+}
+
+const Customize = ({
+    name,
+    visibility,
+    onNameChanged,
+    onVisibilityChanged,
+    ...props
+}: CustomizeProps) => {
+    const {formatMessage} = useIntl();
+
+    return (
+        <div className={props.className}>
+            <div className='name-section-container'>
+                <p>
+                    <strong>
+                        {formatMessage({id: 'worktemplates.customize.name_title', defaultMessage: 'What project is this for?'})}
+                    </strong>
+                </p>
+                <p className='customize-name-text'>
+                    {formatMessage({id: 'worktemplates.customize.name_description', defaultMessage: 'This will help you and others find your project items. You can always edit this later.'})}
+                </p>
+                <input
+                    type='text'
+                    autoFocus={true}
+                    placeholder={formatMessage({id: 'worktemplates.customize.name_input_placeholder', defaultMessage: 'e.g. Web app, Growth, etc.'})}
+                    value={name}
+                    onChange={(e) => onNameChanged(e.target.value)}
+                />
+
+                <Input
+                    label='Test'
+                    value={name}
+                    onChange={(e) => onNameChanged(e.target.value)}
+                    placeholder={formatMessage({id: 'worktemplates.customize.name_input_placeholder', defaultMessage: 'e.g. Web app, Growth, etc.'})}
+                />
+            </div>
+            <div className='visibility-section-container'>
+                <p>
+                    <strong>
+                        {formatMessage({id: 'worktemplates.customize.visibility_title', defaultMessage: 'Who should have access to this?'})}
+                    </strong>
+                </p>
+                <PublicPrivateSelector
+                    selected={visibility === Visibility.Public ? 'O' : 'P'}
```

Comment:
nit: I know this is used this way every where, but a good practice is to avoid the use of magic strings/characters, for these cases a constant/enum is more semantic and the code is more organized

Reply:
Absolutely right, it fell off my radar!